src/app/signin/signin.component.ts

```diff
@@ -7,25 +7,38 @@ import {
 import { Router }            from '@angular/router';
 import { AuthService }       from '../services/auth.service';
 
+import {MatDialog, MAT_DIALOG_DATA} from '@angular/material/dialog';
+import {MpOublieComponent} from '../mp-oublie/mp-oublie.component';
+
+export interface DialogData {
+  animal: 'panda' | 'unicorn' | 'lion';
+}
+
 @Component({
   selector: 'app-signin',
   templateUrl: './signin.component.html',
   styleUrls: ['./signin.component.scss']
 })
 export class SigninComponent implements OnInit {
-  
+
   signinForm: FormGroup;
   errorMessage: string;
-  
+
+
   constructor(private formBuilder: FormBuilder,
               private authService: AuthService,
-              private router: Router) { }
+              private router: Router, public dialog: MatDialog,
+              ) { }
 
-  ngOnInit() {
-    this.initForm();
-  
+  openDialog() {
+    const dialogRef = this.dialog.open(MpOublieComponent, {
+      data: {
```

Comment:
Le data te permet d'envoyer une variable au composant dialog. Tu n'as pas besoin de data dans ce cas => à supprimer.
tu peux faire directement : 
const dialogRef = this.dialog.open(MpOublieComponent);